src/app/clusters/time-format-localization-server/time-format-localization-logic.cpp

```diff
@@ -0,0 +1,250 @@
+    VerifyOrReturnValue(it.IsValid(), false);
+
+    CalendarTypeEnum type;
+    bool found = false;
+
+    while (it.Next(type))
+    {
+        // Update the optional validCalendar to a value from the SupportedList.
+        // This will return either the last element of the list or the requested
+        // value if exists.
+        if (validCalendar != nullptr)
+        {
+            *validCalendar = type;
+        }
+
+        if (type == reqCalendar)
+        {
+            found = true;
+            break;
+        }
+    }
+
+    return found;
+}
+
+TimeFormatLocalization::CalendarTypeEnum TimeFormatLocalizationLogic::GetActiveCalendarType() const
+{
+    return mCalendarType;
+}
+
+DataModel::ActionReturnStatus TimeFormatLocalizationLogic::setHourFormat(TimeFormatLocalization::HourFormatEnum rHour)
+{
+    VerifyOrReturnValue(mAttrProvider != nullptr, Protocols::InteractionModel::Status::Failure);
+
+    if (rHour == HourFormatEnum::kUnknownEnumValue)
+    {
+        return Protocols::InteractionModel::Status::ConstraintError;
+    }
+
+    CHIP_ERROR result = mAttrProvider->WriteValue({ kRootEndpointId, TimeFormatLocalization::Id, HourFormat::Id },
+                                                  { reinterpret_cast<const uint8_t *>(&rHour), sizeof(rHour) });
+    if (result == CHIP_NO_ERROR)
+    {
+        mHourFormat = rHour;
+        return Protocols::InteractionModel::Status::Success;
+    }
+
+    return Protocols::InteractionModel::Status::WriteIgnored;
+}
+
+DataModel::ActionReturnStatus TimeFormatLocalizationLogic::setActiveCalendarType(TimeFormatLocalization::CalendarTypeEnum rCalendar)
+{
+    VerifyOrReturnValue(mAttrProvider != nullptr, Protocols::InteractionModel::Status::Failure);
+    // TODO: Confirm error values for this operation.
+    if (!mFeatures.Has(TimeFormatLocalization::Feature::kCalendarFormat))
+    {
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+
+    // Verify that the requested value is in the supported calendars.
+    if (!IsSupportedCalendarType(rCalendar))
+    {
+        return Protocols::InteractionModel::Status::ConstraintError;
+    }
+
+    // Now try to write the value using the AttributeProvider.
+    CHIP_ERROR result = mAttrProvider->WriteValue(
+        { kRootEndpointId, TimeFormatLocalization::Id, TimeFormatLocalization::Attributes::ActiveCalendarType::Id },
+        { reinterpret_cast<const uint8_t *>(&rCalendar), sizeof(rCalendar) });
+    if (result == CHIP_NO_ERROR)
+    {
+        mCalendarType = rCalendar;
+        return Protocols::InteractionModel::Status::Success;
+    }
+
+    return Protocols::InteractionModel::Status::WriteIgnored;
+}
+
+TimeFormatLocalization::HourFormatEnum TimeFormatLocalizationLogic::GetHourFormat() const
+{
+    return mHourFormat;
+}
+
+CHIP_ERROR TimeFormatLocalizationLogic::Attributes(ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder) const
+{
+    // Ensure capacity just in case
+    ReturnErrorOnFailure(
```

Comment:
use https://github.com/project-chip/connectedhomeip/blob/master/src/app/server-cluster/AttributeListBuilder.h#L36

Generates smaller code and does not need to pre-allocate storage (does so automatically)